pkg/auth/authenticator/password/ldappassword/url.go

```diff
@@ -0,0 +1,141 @@
+package ldappassword
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/go-ldap/ldap"
+)
+
+// LDAPURL holds a parsed RFC 2255 URL
+type LDAPURL struct {
+	// Scheme is ldap or ldaps
+	Scheme string
+	// HostPort is the host:port of the LDAP server
+	HostPort string
+	// The DN of the branch of the directory where all searches should start from
+	BaseDN string
+	// The attribute to search for
+	QueryAttribute string
+	// The scope of the search. Can be either 'one', 'sub', or 'base'
+	Scope int
```

Comment:
> yeah, wrong godoc... it's using ints from the ldap library (which are named, but not typed, LDAP constants)

I'd like named ints to make this idiot proof.  The guys dealing with the ldap libraries live and breath it.  People like me hit it at the edges.  Keep me on the straight, narrow, and idiot-proof